Python/Product/PyDebugAttach/PyDebugAttach.cpp

```diff
@@ -783,7 +783,7 @@ bool LoadAndEvaluateCode(
 
     dictSetItem(globalsDict, "__builtins__", getBuiltins());
     auto size = WideCharToMultiByte(CP_UTF8, 0, filePath, (DWORD)wcslen(filePath), NULL, 0, NULL, NULL);
-    char* filenameBuffer = new char[size];
+    char* filenameBuffer = new char[size + 1];
     if (WideCharToMultiByte(CP_UTF8, 0, filePath, (DWORD)wcslen(filePath), filenameBuffer, size, NULL, NULL) != 0) {
         filenameBuffer[size] = 0;
```

Comment:
According to docs, the different strFromString implementations take a copy of the buffer, which means we'd also be missing a `delete[] filenameBufer` after `dictSetItem`?

Reply:
Looks like it. I would consider this change to be minfix that takes care of the crash - ideally we should re-revert the static analysis fixes, and then do the same thing here (i.e. use `std::string` etc) to avoid all this mess. This would have also caught this bug immediately in debug builds.

Reply:
Yeah, not too concerned about those few bytes leaking for now. Let's do the minfix for 16.1.